cmd/server/storage/mem_storage.go

```diff
@@ -0,0 +1,82 @@
+package storage
+
+import (
+	"errors"
+	"log"
+	"strconv"
+
+	"github.com/Mr-Filatik/go-metrics-collector/cmd/server/entity"
+	"github.com/Mr-Filatik/go-metrics-collector/cmd/server/repository"
+)
+
+type MemStorage struct {
+	repository *repository.MemRepository
+}
+
+func (s *MemStorage) SetRepository(r *repository.MemRepository) {
+
+	s.repository = r
+}
+
+func (s *MemStorage) GetAll() []entity.Metric {
+
+	return s.repository.GetAll()
+}
+
+func (s *MemStorage) Get(t entity.MetricType, n string) (string, error) {
+
+	if t != entity.Gauge && t != entity.Counter {
+		return "0", errors.New("invalid metric type")
```

Comment:
1. В go в случае ошибки первое возвращенное значение считается невалидным (за несколькими редкими исключениями из стандартной библиотеки), потому можно вернуть пустую строку
```suggestion
		return "", errors.New("invalid metric type")
```

2. В ошибки лучше добавлять контекст: например, текст, объясняющий, что именно пошло не так и где